app/queues/dns/create-record-worker.server.ts

```diff
@@ -0,0 +1,142 @@
+import { delay, Worker } from 'bullmq';
+import { redis } from '~/lib/redis.server';
+import { prisma } from '~/db.server';
+import logger from '~/lib/logger.server';
+import { setMonthsFromNow } from '~/lib/domains.server';
+import { createRecord, getChangeStatus } from '~/lib/dns.server';
+import { RecordStatus } from '@prisma/client';
+
+import type { DomainRecord } from '~/lib/domains.server';
+
+export const addDbRecordQueueName = 'add-db-record';
+export const createDnsRecordQueueName = 'create-dns-record';
+export const checkDnsStatusQueueName = 'check-dns-status';
+export const syncDbStatusQueueName = 'sync-db-status';
+
+export const addDbRecordWorker = new Worker<
+  Pick<DomainRecord, 'username' | 'type' | 'name' | 'value'>
+>(
+  addDbRecordQueueName,
+  async (job) => {
+    logger.info('Creating a record in DB');
+    const { username, type, name, value } = job.data;
+
+    if (!username) {
+      throw new Error('Username is not provided');
+    }
+
+    const existingRecords = await prisma.record.count({
```

Comment:
- I'd put in username just for good measure into the where too.
- There is a limit of how many records a user create in sum, should not we check that too?

Reply:
#110 (could evolve it there, or solve that here).

Reply:
@humphd  Thank you for filing and sharing for me. I will update in next PR.

Reply:
@Genne23v so to confirm, you're going to also add `username` in the follow-up?  Seems like an easy win here in this PR.